### PR TITLE
Lerna to Lage

### PR DESCRIPTION
**Pull request by NickGerleman**
This gives us incremental and scoped JS builds, with more intelligent parallelism. We cache 15/17 projects with builds scripts, not yet configuring for react-native-windows of @office-iss/react-native-win32 which have complex outputs. The incremental build of the repo takes ~7s on my machine.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6454)

### CODE REVIEW

File: lage.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    lint: ["build"],
+    'lint:fix': [],
+    test: ["build"],
+    format: [],
+    'format:verify': [],
+    clean: [],
+  },
+  cacheOptions: {
+    outputGlob: [
+      'js/**',
```
Comment (NickGerleman) [inline lage.config.js:19]: I wish we could do this per-package, but package local lage or backfill config doesn't seem to work as far as I can tell

File: packages/E2ETest/just-task.js [hunk 2]
```diff
@@ -5,37 +5,18 @@
...
-  series,
-  argv,
-  tscTask,
-  eslintTask,
-  cleanTask,
-} = require('just-scripts');
+const { task, series, eslintTask, cleanTask } = require('just-scripts');
 const fs = require('fs');
 
 task('clean', cleanTask({ paths: ['dist'] }));
```
Comment (NickGerleman) [inline packages/E2ETest/just-task.js:11]: Can remove clean as well

File: packages/E2ETest/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "extends": "@rnw-scripts/ts-config",
```
Comment (NickGerleman) [inline packages/E2ETest/tsconfig.json:2]: Can delete the tsconfig probably

File: lage.config.js (hunk 1 above)
Comment (NickGerleman) [reply lage.config.js:19]: Yeah, looks like we don't support using package-local backfill config

File: packages/E2ETest/just-task.js [hunk 4]
```diff
@@ -5,37 +5,17 @@
...
 task('prepareBundle', () => {
   const file = 'windows/ReactUWPTestApp/Bundle';
   if (!fs.existsSync(file)) {
     fs.mkdirSync(file);
   }
 });
 
-task('build', series('ts'));
-task('lint', series('eslint'));
+task('lint', series('eslint', tscTask()));
```
Comment (dannyvv) [inline packages/E2ETest/just-task.js:20]: >series [](start = 13, length = 6)

Should we make this parallel?

Comment (dannyvv) [inline packages/E2ETest/just-task.js:20]: >tscTask [](start = 30, length = 7)

Are we sure that our tsc tasks don't need to be in project order? i.e. we don't have any tsc task whose dependencies need dependent projects? 
At least if we make it part of 'lint' we should disable outputting files or at least update the output folder to be a temp folder so

**Review by dannyvv — APPROVED** [review]
:shipit:

File: packages/E2ETest/just-task.js (hunk 4 above)
Comment (NickGerleman) [reply packages/E2ETest/just-task.js:20]: I set tsconfig to noEmit to not output files. 

our TS source doesn't rely on any other local projects, so I think we're fine.

Comment (NickGerleman) [reply packages/E2ETest/just-task.js:20]: Yeah, I can change that.

**Pull request merged**